Apps/Sandcastle/gallery/Particle System Tails.html

```diff
@@ -0,0 +1,260 @@
+
+// Add plane to scene
+var planeOffsetFromParticles = 50;
+var planePosition = Cesium.Cartesian3.fromDegrees(-75.59777, 40.03883, cartographicIncrease + planeOffsetFromParticles);
+var hpr = new Cesium.HeadingPitchRoll(0.0, Cesium.Math.PI_OVER_TWO, 0.0);
+var orientation = Cesium.Transforms.headingPitchRollQuaternion(planePosition, hpr);
+var entity = viewer.entities.add({
+    //Load the Cesium plane model to represent the entity
+    model : {
+        uri : '../../SampleData/models/CesiumAir/Cesium_Air.gltf',
+        scale : 3.5
+    },
+    position : planePosition,
+    orientation : orientation
+});
+viewer.trackedEntity = entity;
+
+// options
+var rocketOptions = {
+    numberOfParticles : 50.0,
+    iterOffset :  0.1,
+    cartographicStep : 0.000001,
+    baseRadius : 0.0005,
+
+    colorOptions : [{
+        minimumRed : 1.0,
+        green : 0.5,
+        minimumBlue : 0.05,
+        alpha : 1.0
+    }, {
+        red : 0.9,
+        minimumGreen : 0.6,
+        minimumBlue : 0.01,
+        alpha : 1.0
+    }, {
+        red : 0.8,
+        green : 0.05,
+        minimumBlue : 0.09,
+        alpha : 1.0
+    }, {
+        minimumRed : 1,
+        minimumGreen : 0.05,
+        blue : 0.09,
+        alpha : 1.0
+    }],
+
+    particleSystems : []
+};
+
+var cometOptions = {
+    numberOfParticles : 100.0,
+    iterOffset : 0.003,
+    cartographicStep : 0.0000001,
+    baseRadius : 0.0005,
+
+    colorOptions : [{
+        red : 0.6,
+        green : 0.6,
+        blue : 0.6,
+        alpha : 1.0
+    }, {
+        red : 0.6,
+        green : 0.6,
+        blue : 0.9,
+        alpha : 0.9
+    }, {
+        red : 0.5,
+        green : 0.5,
+        blue : 0.7,
+        alpha : 0.5
+    }, {
+        red : 0.5,
+        green : 0.5,
+        blue : 0.5,
+        alpha : 0.9
+    }],
+
+    particleSystems : []
+};
+
+// Creating the Particles
+function create(usingOptions, newLocation, color, iteration) {
+    var cartesianOrigPosition = Cesium.Cartographic.toCartesian(newLocation);
+    var emitterModelMatrix = Cesium.Matrix4.fromTranslation(cartesianOrigPosition, new Cesium.Matrix4());
+
+    var iterOffset = iteration;
+    var force = function(particle) {
+        var cartesianPosition = particle.position;
+        var downDir = Cesium.Cartesian3.normalize(cartesianPosition, new Cesium.Cartesian3());
+
+        cartesianPosition.x -= downDir.x;
+        cartesianPosition.y -= downDir.y;
+        cartesianPosition.z -= downDir.z;
+
+        var cartographicPosition = Cesium.Cartographic.fromCartesian(cartesianPosition);
+
+        var angle = Cesium.Math.PI * 2.0 * iterOffset / usingOptions.numberOfParticles;
+        iterOffset += usingOptions.iterOffset;
+        cartographicPosition.longitude += Math.cos(angle) * usingOptions.cartographicStep;
+        cartographicPosition.latitude += Math.sin(angle) * usingOptions.cartographicStep;
+
+        cartesianPosition = Cesium.Cartographic.toCartesian(cartographicPosition);
```

Comment:
Is it possible to do these calculations on `Cartesian3`s to avoid the conversions back and forth an creating new objects each update?

Reply:
Also these particle paths come off very smooth, can we introduce a little noise with some randomness?